app/src/main/java/com/infomaniak/mail/receivers/NotificationActionsReceiver.kt

```diff
@@ -144,15 +146,17 @@ class NotificationActionsReceiver : BroadcastReceiver() {
         payload: NotificationPayload,
     ) = with(payload) {
 
-        notificationUtils.showMessageNotification(
-            notificationManagerCompat = notificationManagerCompat,
-            payload = payload.apply {
-                behavior = NotificationBehavior(
-                    type = NotificationType.UNDO,
-                    behaviorTitle = context.getString(undoNotificationTitle),
-                )
-            },
-        )
+        globalCoroutineScope.launch {
```

Comment:
as we're already having the scope passed to the showMessageNotificationMethod, couldn't we just factorize the launch inside it ?